openscap_daemon/evaluation_spec.py

```diff
@@ -349,6 +349,21 @@ def get_cpe_ids(self, config):
 
         return cpe_ids
 
+    def select_profile_by_suffix(self, ssg_sds, profile_suffix):
+        profiles = oscap_helpers.get_profile_choices_for_input(ssg_sds, None)
+        for p in profiles:
+            if p.endswith(profile_suffix):
+                self.profile_id = p
+                break
+        else:
+            raise RuntimeError(
+                "Profile with id='%s' doesn't exist on target '%s'. "
```

Comment:
Yeah, throw in EvaluationSpec, then catch it in oscapd-evaluate to make a nice specific message.